mysql/infoschema.go

```diff
@@ -0,0 +1,357 @@
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Couldn't get spanner table : %s", err))
+			continue
+		}
+		spCols, err := internal.GetSpannerCols(conv, srcTable, srcCols)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Couldn't get spanner columns for table %s : err = %s", t.name, err))
+			continue
+		}
+		spSchema, ok2 := conv.SpSchema[spTable]
+		if !ok2 {
+			conv.Stats.BadRows[srcTable] += conv.Stats.Rows[srcTable]
+			conv.Unexpected(fmt.Sprintf("Can't get schemas for table %s", srcTable))
+			continue
+		}
+		// Make a slice for the values
+		v := make([]sql.RawBytes, len(srcCols))
+		scanArgs := make([]interface{}, len(v))
+		for i := range v {
+			scanArgs[i] = &v[i]
+		}
+		// Fetch rows
+		for rows.Next() {
+			// get RawBytes from data
+			err = rows.Scan(scanArgs...)
+			if err != nil {
+				conv.Unexpected(fmt.Sprintf("Couldn't process sql data row: %s", err))
+				conv.StatsAddBadRow(srcTable, conv.DataMode())
+				continue
+			}
+			values := valsToStrings(v)
+			ProcessDataRow(conv, srcTable, srcCols, srcSchema, spTable, spCols, spSchema, values)
+		}
+	}
+}
+
+// Building list of column names to support spatial datatypes instead of
+// using 'SELECT *'. So spatial columns will be fetched using ST_AsText().
+func buildColNameList(srcSchema schema.Table, srcColName []string) string {
+	var srcColTypes []string
+	var colList, colTmpName string
+	for _, colName := range srcColName {
+		// To handle cases where column name is reserved keyword or having space between words
+		colTmpName = "`" + colName + "`"
+		srcColTypes = append(srcColTypes, srcSchema.ColDefs[colName].Type.Name)
+		for _, spatial := range MysqlSpatialDataTypes {
+			if strings.Contains(strings.ToLower(srcSchema.ColDefs[colName].Type.Name), spatial) {
+				colTmpName = "ST_AsText" + "(" + colTmpName + ")" + colTmpName
+				break
+			}
+		}
+		colList = colList + colTmpName + ","
+	}
+	return colList[:len(colList)-1]
+}
+
+// SetRowStats populates conv with the number of rows in each table.
+func SetRowStats(conv *internal.Conv, db *sql.DB) {
+	tables, err := getTables(db)
+	if err != nil {
+
+		conv.Unexpected(fmt.Sprintf("Couldn't get list of table: %s", err))
+		return
+	}
+	for _, t := range tables {
+		// MYSQL schema and name can be arbitrary strings.
+		// Ideally we would pass schema/name as a query parameter,
+		// but MYSQL doesn't support this. So we quote it instead.
+		q := fmt.Sprintf(`SELECT COUNT(*) FROM %s.%s;`, t.schema, t.name)
+		tableName := buildTableName(t.schema, t.name)
+		rows, err := db.Query(q)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Couldn't get number of rows for table %s", tableName))
+			continue
+		}
+		defer rows.Close()
+		var count int64
+		if rows.Next() {
+			err := rows.Scan(&count)
+			if err != nil {
+				fmt.Printf("Can't get row count: %s\n", err)
+				continue
+			}
+			conv.Stats.Rows[tableName] += count
+		}
+	}
+}
+
+type schemaAndName struct {
```

Comment:
should we define the struct at the beginning of the file?

Reply:
Considering their previous comments on PR, at couple of instances they had told me to move variable declarations just above where they are used. Hence, I have kept it just above the function where it is used.

Reply:
alright, then no issue